pybamm/discretisations/base_discretisation.py

```diff
@@ -200,15 +199,22 @@ def process_symbol(self, symbol, y_slices=None, boundary_conditions={}):
 
             return new_symbol
 
+        elif isinstance(symbol, pybamm.Scalar):
+            return pybamm.Scalar(symbol.value, domain=symbol.domain)
+
+        elif isinstance(symbol, pybamm.Array):
+            return symbol.__class__(symbol.entries, domain=symbol.domain)
```

Comment:
```suggestion
            return symbol.Array(symbol.entries, domain=symbol.domain)
```

Reply:
not sure what is better, but should be consistent with Scalar elif above

Reply:
`Array` can be either `Vector` or `Matrix`, would be better to return the same type I think

Reply:
ok, forgot about that. Ignore that suggestion then!